### PR TITLE
CXP-1303: Display selection criteria per attribute groups

### PR DESCRIPTION
**Pull request by RiffVentura**
Added attribute group code and label to the payload of the internal attribute endpoints to bucket attributes on the front end
image

### CODE REVIEW

**Review by eoliveauakeneo — APPROVED** [review]

File: components/catalogs/back/tests/Integration/IntegrationTestCase.php [hunk 1]
```diff
@@ -525,6 +524,29 @@ private function createAttributeOptions(AttributeInterface $attribute, array $co
         self::getContainer()->get('pim_catalog.saver.attribute_option')->saveAll($options);
     }
 
+    /**
+     *     * @param array{
```
Comment (masacc) [inline components/catalogs/back/tests/Integration/IntegrationTestCase.php:528]:
```suggestion
     * @param array{
```

File: components/catalogs/back/tests/Integration/IntegrationTestCase.php [hunk 2]
```diff
@@ -525,6 +524,29 @@ private function createAttributeOptions(AttributeInterface $attribute, array $co
...
+     * } $data
+     */
+    protected function createAttributeGroup(array $data): void
+    {
+        $attributeGroup = self::getContainer()->get('pim_catalog.repository.attribute_group')
+            ->findOneByIdentifier($data['code']);
+
+        if (null === $attributeGroup) {
+            $attributeGroup = self::getContainer()->get('pim_catalog.factory.attribute_group')->create();
+            ;
```
Comment (masacc) [inline components/catalogs/back/tests/Integration/IntegrationTestCase.php:542]: rm

File: components/catalogs/front/src/components/ProductSelection/components/AddCriterionDropdown.tsx [hunk 3]
```diff
@@ -66,15 +68,18 @@ const AddCriterionDropdown: FC<{}> = () => {
...
-                                {translate('akeneo_catalogs.product_selection.add_criteria.section_attributes')}
-                            </Dropdown.Section>
-                        )}
-                        {attributesCriterionFactories?.map(factory => (
-                            <Dropdown.Item key={factory.id} onClick={() => handleNewCriterion(factory.factory())}>
-                                {factory.label}
-                            </Dropdown.Item>
+                        {attributesCriterionFactoriesByAttributeGroup.map(groupedAttributes => (
+                            <div key={groupedAttributes.code}>
+                                <Dropdown.Section>{groupedAttributes.label}</Dropdown.Section>
```
Comment (masacc) [inline components/catalogs/front/src/components/ProductSelection/components/AddCriterionDropdown.tsx:73]: When searching, does it removes the `<Dropdown.Section>` if there is no attribute in it?

**Review by masacc — APPROVED** [review]

File: components/catalogs/front/src/components/ProductSelection/components/AddCriterionDropdown.tsx (hunk 3 above)
Comment (RiffVentura) [reply components/catalogs/front/src/components/ProductSelection/components/AddCriterionDropdown.tsx:73]: Yes as it's not possible to have a group without attributes in it (the hook that buckets attribute together ensures that). Each attribute belongs to a group

**Pull request merged**